Mgt/localization + right-to-left (#10569)
microsoftgraph/microsoft-graph-docs: concepts/toolkit/localization.md

---
title: "Localizing the Microsoft Graph Toolkit"
description: "Use LocalizationHelper to translate component strings"
localization_priority: Normal
author: vogtn
---

# Localizating components in the Microsoft Graph Toolkit

Each component and subsequent folder contains it's own set of strings that can be
localized:

`packages/mgt/src/components/mgt-people-picker/strings.ts`

```ts
export const strings = {
  inputPlaceholderText: "Start typing a name",
  noResultsFound: `We didn't find any matches.`,
  loadingMessage: "Loading...",
};
```

The component itself then must use the BaseComponent's `string` property which is updated if the `string` property of LocalizationHelper is updated.

`packages/mgt/src/components/mgt-people-picker/mgt-people-picker.ts`

```html
<div label="loading-text" aria-label="loading" class="loading-text">
  ${this.strings.loadingMessage}
</div>
```

> 

Reviewer: Move this down after the example on how localization works and lead in with an overview of Localization and why it's important.

Author: Wrote a short intro here:
9465394

I tried to base it on this context:
https://docs.microsoft.com/en-us/dotnet/standard/globalization-localization/localization

## Using LocalizationHelper to translate strings

To access the LocalizationHelper you must import the module from the package and utilze the `string` property.

The strings can be referenced at a global level or per component (with the `_components:` selector), but global assignment of common properties will cascade unless further specified.

example:

```html

> <script type="module">

Reviewer: I think you can remove the script tags, they don't add anything to the example

Author: Removed:
9465394

Only thing, it removes some of the highlighting on the object inside the code block.

>   import { LocalizationHelper } from "./packages/mgt/dist/es6/index.js";

Reviewer: Think about how an external developer would import the toolkit, not how we import it for testing

```suggestion
  import { LocalizationHelper } from "@microsoft/mgt";
```

Author: Good point, updated: 

9465394

---

> title: "Localizing the Microsoft Graph Toolkit"

Reviewer: ```suggestion
title: "Localizing the Microsoft Graph Toolkit Components"
```

> description: "Use LocalizationHelper to translate component strings"

Reviewer: ```suggestion
description: "Use LocalizationHelper to localize Microsoft Graph Toolkit Components"
```

localization_priority: Normal
author: vogtn
---

> # Localizing the Microsoft Graph Toolkit

Reviewer: ```suggestion
# Localizing the Microsoft Graph Toolkit Components
```

Localization is a important aspect of application development to support users with various language requirements globally.

> The Microsoft Graph Toolkit components can localize designated strings so the user-interface reflects the desiered linguistic experience.

Reviewer: ```suggestion
The Microsoft Graph Toolkit components can be localized so the user-interface reflects the desired linguistic experience.
```

> ## Using LocalizationHelper to translate strings
> 
> To access the LocalizationHelper you must import the module from the package and utilze the `string` property.

Reviewer: ```suggestion
## Use LocalizationHelper to add localized strings

All strings in the toolkit are not localized, but you can provide your own localized strings and manage different languages through the same process you use for localizing your app. To facilitate localization, the toolkit exposes the `LocalizationHelper` static class:
```

The strings can be referenced at a global level or per component (with the `_components:` selector), but global assignment of common properties will cascade unless further specified.

example:

```ts
import { LocalizationHelper } from "@microsoft/mgt";

LocalizationHelper.strings = {
  noResultsFound: "لم نجد أي قنوات",
  _components: {
    login: {
      signInLinkSubtitle: "login",
      signOutLinkSubtitle: "خروج",
    },
    "people-picker": {
      inputPlaceholderText: "ابدأ في كتابة الاسم",
      noResultsFound: "لم نجد أي قنوات", //collision with global defined noResultsFound will overwrite with local result
      loadingMessage: "...جار التحميل",
    },
    "teams-channel-picker": {
      inputPlaceholderText: "حدد قناة",
      noResultsFound: "local NoResultsFound Example",
      // loadingMessage: is default string "Loading..." for this example since not defined globally or locally
    },
    tasks: {
      removeTaskSubtitle: "delete",
      cancelNewTaskSubtitle: "canceltest",
      newTaskPlaceholder: "newTaskTest",
      addTaskButtonSubtitle: "addme",
    },
  },
};
```

> ## Localizating components in the Microsoft Graph Toolkit

Reviewer: Instead of this section, what if we list all strings for all components. For examples

## Strings

### Agenda

```ts
{
  "key": "value",
  "key": "value
}
```

### People Picker

```ts
{
  "key": "value",
  "key": "value
}
```

Author: Thanks great suggestion!

Author: Updated with the current available strings, will add the person-card strings in a future PR: 

f87c41a
